api/common/interfaces.go

```diff
@@ -647,91 +647,152 @@ func getSubscribers(s Serverer, params map[string]interface{}) map[string]interf
 		return respPacking(INVALID_PARAMS, "topic should be a string")
 	}
 
-	bucket, ok := params["bucket"].(float64)
+	var bucket float64
+	if _, ok := params["bucket"]; ok {
+		bucket, ok = params["bucket"].(float64)
```

Comment:
Is there a reason we are not checking uint32 here? Also do we keep bucket because of compatibility?

Reply:
when you unmarshal json into `map[string]interface{}` all numbers become float64. I can check that it falls within uint32 range or even into buckets count limit, it's just how uint32 is treated all over `interfaces.go` and I don't know if it's really necessary. Ideally every method should have their own request/response struct so that it'll be unmarshalled correctly or not unmarshalled at all.

yes, bucket are kept because there will still be some subscriptions in other buckets when we roll it out. but we should remove them from here sometimes in the future (when all non-zero bucket subscriptions are expired and new ones can't be created anymore)

Reply:
I see. BTW I don't think anyone is using nonzero bucket AFAIK. Seems that everyone is just hard coding bucket=0 now 😂

Reply:
I have height limit after which it won't accept non-zero buckets anymore. If at that point there's no non-zero buckets in statedb, then we can safely remove that.